tests/internal/tls/validFor_test.go

```diff
@@ -0,0 +1,162 @@
+package tls_test
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"testing"
+
+	"github.com/openziti/identity"
+	ziti_tls "github.com/openziti/ziti/internal/tls"
+)
+
+// mockIdentity implements the Identity interface for testing
+type mockIdentity struct {
+	serverCerts []*tls.Certificate
+	clientCert  *tls.Certificate
+}
+
+func (m *mockIdentity) Cert() *tls.Certificate         { return m.clientCert }
+func (m *mockIdentity) ServerCert() []*tls.Certificate { return m.serverCerts }
+func (m *mockIdentity) CA() *x509.CertPool             { return nil }
+func (m *mockIdentity) CaPool() *identity.CaPool       { return nil }
+func (m *mockIdentity) ServerTLSConfig() *tls.Config   { return nil }
+func (m *mockIdentity) ClientTLSConfig() *tls.Config   { return nil }
+func (m *mockIdentity) Reload() error                  { return nil }
+func (m *mockIdentity) WatchFiles() error              { return nil }
+func (m *mockIdentity) StopWatchingFiles()             {}
+func (m *mockIdentity) SetCert(pem string) error       { return nil }
+func (m *mockIdentity) SetServerCert(pem string) error { return nil }
+func (m *mockIdentity) GetConfig() *identity.Config    { return nil }
+func (m *mockIdentity) ValidFor(address string) bool   { return true }
+
+const (
+	validDNS   = "example.com"
+	invalidDNS = "invalid.com"
+	validIP4   = "192.168.1.1"
+	invalidIP4 = "10.0.0.1"
+	validIP6   = "::1"
```

Comment:
I would expect to see an ipv6 compress vs expanded test. These two ipv6 addresses are equivalent:

- 2001:0db8:0000:0000:0000:ff00:0042:8329
- 2001:db8::ff00:42:8329

Reply:
added these as tests. see https://github.com/openziti/identity/pull/60/files#diff-1c7a84c01de0f235cdda7cd9434265796e328b9b335eaf138a96a9df29c78769R163-R197